### PR TITLE
chore: refactor spelling test, add plugins & fix spellings

### PR DESCRIPTION
**Pull request by jkodu**
- use multiple english language helper plugins to improve spell check
- fix spelling mistakes
- refactor

Closes issue(s):
- NA

### CODE REVIEW

File: _rules/iframe-accessible-name-cae760.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Each target element has an [accessible name][] that is not empty (`""`).
 
 ## Assumptions
 
-- The rule assumes that the target `iframe` is used as an [user interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components).
+- The rule assumes that the target `iframe` is used as a [user interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components).
```
Comment (Jym77) [inline _rules/iframe-accessible-name-cae760.md]: I kinda disagree with that change 😕 
I assume that the an/a checker is thrown out by the square bracket.

File: _rules/image-filename-as-accessible-name-9eb3f6.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ The `img` element's [accessible name][] uses the filename which accurately descr
 
 #### Passed Example 2
 
-The `img` element's [accessible name][] includes the filename, which in combination with the text content of the `a` element accurately describes the image.
+The `img` element's [accessible name][] includes the filename, which in combination with the text content of the `<a>` element accurately describes the image.
```
Comment (Jym77) [inline _rules/image-filename-as-accessible-name-9eb3f6.md]: Nope. We don't put element name in brackets.
I assume the an/a checker is thrown out by the "a element" look-alike…

File: _rules/link-accessible-name-c487ae.md [hunk 3]
```diff
@@ -158,7 +158,7 @@ When `link` is off screen.
 
 #### Passed Example 11
 
-`a` element where [accessible name][] is not empty.
+`<a>` element where [accessible name][] is not empty.
```
Comment (Jym77) [inline _rules/link-accessible-name-c487ae.md]: No.

File: _rules/link-accessible-name-c487ae.md [hunk 4]
```diff
@@ -247,7 +247,7 @@ Link is completely empty, but still shows up in focus order, so it should have a
 
 #### Failed Example 10
 
-`a` element where [accessible name][] through content is empty.
+`<a>` element where [accessible name][] through content is empty.
```
Comment (Jym77) [inline _rules/link-accessible-name-c487ae.md]: No.

File: _rules/links-identical-name-equivalent-purpose-b20e66.md [hunk 5]
```diff
@@ -60,7 +60,7 @@ This rule assumes that assistive technologies are exposing all links on the page
 
 #### Passed Example 1
 
-A set of two HTML `a` elements have the same [accessible name][] and link to the [same resource][].
+A set of two HTML `<a>` elements have the same [accessible name][] and link to the [same resource][].
```
Comment (Jym77) [inline _rules/links-identical-name-equivalent-purpose-b20e66.md]: No.

File: _rules/links-identical-name-equivalent-purpose-b20e66.md [hunk 6]
```diff
@@ -69,7 +69,7 @@ A set of two HTML `a` elements have the same [accessible name][] and link to the
 
 #### Passed Example 2
 
-A set of two HTML `a` elements that both resolve to [same resource][] after instant redirect.
+A set of two HTML `<a>` elements that both resolve to [same resource][] after instant redirect.
```
Comment (Jym77) [inline _rules/links-identical-name-equivalent-purpose-b20e66.md]: No.

File: _rules/links-identical-name-equivalent-purpose-b20e66.md [hunk 7]
```diff
@@ -96,7 +96,7 @@ These two links go to pages where the content section is the same, but where the
 
 #### Passed Example 5
 
-These two HTML `a` elements link to URLs that differ due to trailing slashes, but resolve to the [same resource][] after redirects caused by user agent.
+These two HTML `<a>` elements link to URLs that differ due to trailing slashes, but resolve to the [same resource][] after redirects caused by user agent.
```
Comment (Jym77) [inline _rules/links-identical-name-equivalent-purpose-b20e66.md]: No.

File: _rules/links-identical-name-equivalent-purpose-b20e66.md [hunk 8]
```diff
@@ -145,7 +145,7 @@ These two HTML `span` element have an [explicit role][] of link, and lead to t
 
 #### Passed Example 9
 
-A set of two SVG `a` elements have the same [accessible name][] and link to the [same resource][].
+A set of two SVG `<a>` elements have the same [accessible name][] and link to the [same resource][].
```
Comment (Jym77) [inline _rules/links-identical-name-equivalent-purpose-b20e66.md]: No.

File: _rules/links-identical-name-equivalent-purpose-b20e66.md [hunk 9]
```diff
@@ -163,7 +163,7 @@ A set of two SVG `a` elements have the same [accessible name][] and link to the
 
 #### Passed Example 10
 
-A set of one SVG `a` element and one HTML `a` element that have the same [accessible name][] and link to the [same resource][].
+A set of one SVG `<a>` element and one HTML `<a>` element that have the same [accessible name][] and link to the [same resource][].
```
Comment (Jym77) [inline _rules/links-identical-name-equivalent-purpose-b20e66.md]: No. No.

File: _rules/links-identical-name-equivalent-purpose-b20e66.md [hunk 10]
```diff
@@ -266,7 +266,7 @@ The same [accessible name][] is used for image links going to different resource
 
 #### Failed Example 5
 
-A set of two SVG `a` elements that have the same [accessible name][] but link to different resources.
+A set of two SVG `<a>` elements that have the same [accessible name][] but link to different resources.
```
Comment (Jym77) [inline _rules/links-identical-name-equivalent-purpose-b20e66.md]: No.

File: _rules/iframe-accessible-name-cae760.md (hunk 1 above)
Comment (Jym77) [reply _rules/iframe-accessible-name-cae760.md]: After double checking, "a user" is correct ("a you-zer", not "an uh-zer"). My bad.

**Comment by jkodu** [comment]
@Jym77 as much as I would have not liked to change `a` to `<a>`, the spell checker is going bust on this one.

Any bright ideas?

**Comment by Jym77** [comment]
> @Jym77 as much as I would have not liked to change `a` to `<a>`, the spell checker is going bust on this one.
> 
> Any bright ideas?

Disable the a/an part of retext by default. For us, it's causing more harm than good.

Possibly run it locally every so often to catch the actual issues. Seeing how few there are even though heavy work has happened for 2 years here, I say it should be OK.

**Comment by jkodu** [comment]
After several attempts to curate the markdown of all `backtick` enclosed text etc., and passing it to spelling tests. All of them failed for making the `indefinite-artcile` (usage of `an` and `a`) happy.

Last resort, I removed the plugin `indefinite-article`.

This change has helped us catch several other spelling mistakes.

File: _rules/link-accessible-name-c487ae.md (hunk 4 above)
Comment (Jym77) [inline _rules/link-accessible-name-c487ae.md]:
```suggestion
`a` element where [accessible name][] through content is empty.
```

**Comment by jkodu** [comment]
Given this work is `chore`, we should be ok with 2 approvals on this.

**Pull request merged**